### PR TITLE
ingress: allow wildcard hosts in IngressRule

### PR DESCRIPTION
**Pull request by cmluciano**
Signed-off-by: Christopher M. Luciano <cmluciano@us.ibm.com>

**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:
This PR tests that wildcard hosts are accepted and updates the types documentation to explicitly state the support of wildcard hosts

**Which issue(s) this PR fixes**: 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
ingress: allow wildcard hosts in IngressRule
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

[Ingress KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190125-ingress-api-group.md)

### CODE REVIEW

**Comment by cmluciano** [comment]
/assign @robscott @liggitt 

This should be the final PR for the planned 1.18 Ingress changes

File: pkg/apis/networking/types.go [hunk 1]
```diff
@@ -349,18 +349,27 @@ type IngressStatus struct {
 // host match, then routed to the backend associated with the matching
 // IngressRuleValue.
 type IngressRule struct {
-	// Host is the fully qualified domain name of a network host, as defined
-	// by RFC 3986. Note the following deviations from the "host" part of the
-	// URI as defined in the RFC:
-	// 1. IPs are not allowed. Currently an IngressRuleValue can only apply to the
-	//	  IP in the Spec of the parent Ingress.
+	// Note the following deviations from the "host" part of the
```
Comment (liggitt) [inline pkg/apis/networking/types.go]: nit: keep the first sentence of the godoc beginning with the field name to match conventions

**Review by liggitt — COMMENTED** [review]
one doc nit, lgtm otherwise

**Pull request merged**